ear-production-suite-plugins/lib/include/helper/eps_to_ear_metadata_converter.hpp

```diff
@@ -26,12 +30,14 @@ struct EpsToEarMetadataConverter {
 
   static std::vector<ear::DirectSpeakersTypeMetadata> convert(
       const proto::DirectSpeakersTypeMetadata &epsMetadata) {
-    auto packFormatId =
-        proto::SpeakerLayoutTranslator::proto2ear(epsMetadata.layout());
+    std::stringstream ss;
+    ss << "AP_0001" << std::setw(4) << std::setfill('0') << std::hex
+       << epsMetadata.packformatidvalue();
+    std::string audioPackFormatID{ ss.str() };
```

Comment:
Interesting - will tidy this up here and in all the other places were constructing IDs with stringstream.

My own tests for 1 million ID generates;

- **Existing (stringstream)**: 725ms (!)
- **Via ID construct and formatId**: 31ms
- **Via adm::detail::formatHex**: 7ms

Although formatHex is fastest, we'll go with formatId since - as pointed out - ID construction will pick up validation issues earlier if it ever starts going wrong!